src/Adapter/MSTestAdapter.PlatformServices/Execution/TestMethodInfo.cs

```diff
@@ -394,6 +394,11 @@ private async Task<TestResult> ExecuteInternalAsync(object?[]? arguments, Cancel
         {
             try
             {
+                foreach ((MethodInfo method, TimeoutInfo? timeoutInfo) in Parent.Parent.GlobalTestInitializations)
```

Comment:
It's documented by the tests but I would make it also explicit in the code that we want to run the global init/cleanup before and after everything. It's not like a regular init/cleanup.